retail/3.2/modules/ROOT/pages/rcg-sfdc-customization-guide.adoc

```diff
@@ -0,0 +1,453 @@
+
+== Create Platform Events and Apex Triggers
+
+The following Platform Event Object and Apex Triggers must be created in order to capture updates from Salesforce. The scripts to create these Apex Triggers and add fields to Platform Event Object are located in the https://anypoint.mulesoft.com/exchange/2cc3c2c9-ddd3-4582-bdcc-b510f30065a7/rcg-common-resources/[RCG Common Resources] project.
+
+|===
+| Salesforce Object | Salesforce Platform Event Object | Salesforce Platform Event API Name | Apex Trigger Name
+
+| Account
+| Account
+| Account__e
+| AccountEventsPub
+|===
+
+=== Create Platform Event Object
+
+This implementation requires `Platform Event` to be created in Salesforce. Follow these steps to create a `Platform Event` object:
+
+. Log in to Salesforce.
+. Click *Setup* and search for "Platform Events" in the *Quick Find* box. The Platform Events page shows with a list of available Platform Events.
+. Click the *New Platform Event* button to create an Object. Complete the *Platform Information Details* fields, including *Label*, *Plural Label*, *Object with Salesforce Object*, and *Salesforce Platform Event API Name* as indicated in the previous table.
+. Click *Save*.
+
+. To add Custom Fields & RelationShips to Platform Event Definitions, run the utility script `CreatePlatformEventsCustomFields.txt` available in the `rcg-common-resources/salesforce` folder of https://anypoint.mulesoft.com/exchange/2cc3c2c9-ddd3-4582-bdcc-b510f30065a7/rcg-common-resources/[RCG Common Resources] project.
+. Follow the steps to run the script provided in `Create via Apex Script` under `Custom Fields`.
+
+=== Create Apex Triggers
+
+Follow these steps to create the Apex Triggers:
+
+. Log in to Salesforce.
+. Click Setup to search for Object Manager in Quick Find Box.
+. Search for the Salesforce object. This opens a page that shows the details of the object.
+. Click *Triggers* in the Details section to view a list of available triggers for the object.
+. Click *New*.
+. Search for the Apex Trigger name, which is indicated in the previous table. Copy the Trigger Script code `CreateApexTriggers.txt` available in the `rcg-common-resources/salesforce` folder of https://anypoint.mulesoft.com/exchange/2cc3c2c9-ddd3-4582-bdcc-b510f30065a7/rcg-common-resources/[RCG Common Resources] project and paste it into the script editor.
+. Obtain the `LastModifiedById` by running the query from the Additional Notes section.
+. Replace the values in the script and save the trigger.
+. Click *Save* and ensure the *isActive* checkbox is checked.
+
+=== Additional Notes
+
+* The Apex Trigger that is created filters records based on `LastModifiedBy <> {Service account ID}` to avoid cyclical updates performed by the Salesforce Customers System API. Run the query `SELECT Id FROM User WHERE Username='<service-account-username>'` in the Developer Console to obtain Service account ID.
```

Comment:
```suggestion
The Apex Trigger that is created filters records based on `LastModifiedBy <> {Service account ID}` to avoid cyclical updates performed by the Salesforce Customers System API. Run the query `SELECT Id FROM User WHERE Username='<service-account-username>'` in the Developer Console to obtain Service account ID.
```